### PR TITLE
docs(canvas): add documentation for x_bounds, y_bounds

### PR DESCRIPTION
**Pull request by orhun**
> Upstream: [#666](https://github.com/fdehau/tui-rs/pull/666)

## Description

There is no description in docs for `Canvas::x_bounds` and `Canvas::y_bounds`, but there is a [question](https://github.com/fdehau/tui-rs/issues/286) about that with response of the author. So this PR puts this response as documentation for the methods with some small corrections and formatting.

## Testing guidelines

This PR does not add any code, but all code fragments in documentation are valid & buildable.

## Checklist

* [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
* [x] I have added relevant tests.
* [x] I have documented all new additions.

### CODE REVIEW

File: src/widgets/canvas/mod.rs [hunk 1]
```diff
@@ -384,11 +384,65 @@ where
...
+    ///     .paint(|ctx| {
+    ///         ctx.draw(&Map {
+    ///             color: Color::White,
+    ///             resolution: MapResolution::High,
+    ///         });
+    ///     })
+    ///     .x_bounds([-180.0, 180.0])
+    ///     .y_bounds([-90.0, 90.0]);
+    /// ```
+    ///
```
Comment (mindoodoo) [inline src/widgets/canvas/mod.rs:443]: Since both additions are pretty much the same, maybe we could avoid some wordiness by linking to the ```x_bounds``` ?

File: src/widgets/canvas/mod.rs [hunk 2]
```diff
@@ -384,11 +384,65 @@ where
...
+    /// If you were to "zoom" to a certain part of the world you may want to choose different
+    /// bounds.
```
Comment (mindoodoo) [inline src/widgets/canvas/mod.rs:445]: Maybe not really needed ? A bit wordy

File: src/widgets/canvas/mod.rs [hunk 3]
```diff
@@ -384,11 +384,65 @@ where
...
```
Comment (mindoodoo) [inline src/widgets/canvas/mod.rs:413]: Maybe not really need ? A bit wordy.

**Review by mindoodoo — CHANGES_REQUESTED** [review]
I feel like we don't really need the examples in the end. If the issue is ever brought up again we can add them back in, but for the time being we could simplify this

File: src/widgets/canvas/mod.rs (hunk 3 above)
Comment (sayanarijit) [inline src/widgets/canvas/mod.rs:413]:
```suggestion
    /// Define the viewport of the canvas.
    /// If you were to "zoom" to a certain part of the world you may want to choose different
    /// bounds.
```

File: src/widgets/canvas/mod.rs [hunk 4]
```diff
@@ -384,11 +384,65 @@ where
...
+    ///         ctx.draw(&Map {
+    ///             color: Color::White,
+    ///             resolution: MapResolution::High,
+    ///         });
+    ///     })
+    ///     .x_bounds([-180.0, 180.0])
+    ///     .y_bounds([-90.0, 90.0]);
+    /// ```
+    ///
+    /// If you were to "zoom" to a certain part of the world you may want to choose different
```
Comment (sayanarijit) [inline src/widgets/canvas/mod.rs:444]:
```suggestion
    /// Define the viewport of the canvas.
    ///
    /// If you were to "zoom" to a certain part of the world you may want to choose different
```

**Review by sayanarijit — COMMENTED** [review]
@mindoodoo @orhun Feel free to merge the suggested changes if you agree with it.
